### PR TITLE
fix checks for gcc disable warning flags

### PR DESCRIPTION
**Pull request by bruce-richardson**
GCC does not print a warning or error for unknown options if the options
are to disable warnings. Therefore, when checking for options starting
'-Wno-', also check the opposite enabling option.  This fixes the case
where e.g.  -Wno-implicit-fallthrough is incorrectly reported as supported
by gcc 5.4.

### CODE REVIEW

File: mesonbuild/compilers/c.py [hunk 1]
```diff
@@ -713,6 +713,12 @@ def thread_link_flags(self):
         return ['-pthread']
 
     def has_multi_arguments(self, args, env):
+        for arg in args[:]:
+            # some compilers, e.g. GCC, don't warn for unsupported warning-disable
+            # flags, so when we are testing a flag like "-Wno-forgotten-towel", also
+            # check the equivalent enable flag too "-Wforgotten-towel"
+            if arg.startswith('-Wno-'):
+                args.append(arg.replace('-Wno-','-W'))
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/c.py]: __[flake8]__
* [E231] missing whitespace after ','

**Comment by jpakkane** [comment]
Please rebase so CI gets run. It is now back in working condition again.

**Comment by bruce-richardson** [comment]
done. Here's hoping for successful results!

**Comment by nirbheek** [comment]
Test failure is relevant. Do the tests pass for you on your machine?

**Comment by jpakkane** [comment]
ping

**Comment by bruce-richardson** [comment]
I'm looking at test failures now. Agree some seem relevant.

File: mesonbuild/compilers/c.py [hunk 2]
```diff
@@ -744,6 +744,11 @@ def thread_link_flags(self):
         return ['-pthread']
 
     def has_multi_arguments(self, args, env):
+        # some compilers, e.g. GCC, don't warn for unsupported warning-disable
+        # flags, so when we are testing a flag like "-Wno-forgotten-towel", also
+        # check the equivalent enable flag too "-Wforgotten-towel"
+        if len(args) == 1 and args[0].startswith('-Wno-'):
+                args.append(args[0].replace('-Wno-', '-W'))
```
Comment (keszybz) [inline mesonbuild/compilers/c.py]: `replace` goes over the whole string. This would be better written as `'-W' + args[0][5:]`.

**Comment by bruce-richardson** [comment]
reworked as suggested

**Comment by keszybz** [comment]
LGTM.

**Comment by nirbheek** [comment]
I rebased the branch, added a test, and made it work for multiple `-Wno` arguments. I'll merge this once the tests pass.

**Pull request merged**